crates/ika-core/src/dwallet_mpc/mpc_protocols/dwallet_dkg.rs

```diff
@@ -6,15 +6,74 @@ use dwallet_mpc_types::dwallet_mpc::{
     VersionedPublicKeyShareAndProof,
 };
 use ika_types::dwallet_mpc_error::{DwalletMPCError, DwalletMPCResult};
-use ika_types::messages_dwallet_mpc::AsyncProtocol;
+use ika_types::messages_dwallet_mpc::{
+    AsyncProtocol, DWalletDKGFirstRoundRequestEvent, DWalletDKGSecondRoundRequestEvent,
+    DWalletImportedKeyVerificationRequestEvent, DWalletSessionEvent, MPCProtocolInitData,
+    SessionInfo,
+};
 use mpc::Party;
 use twopc_mpc::dkg::Protocol;
 /// This struct represents the initial round of the DKG protocol.
-pub type DKGFirstParty = <AsyncProtocol as Protocol>::EncryptionOfSecretKeyShareRoundParty;
-pub(super) type DWalletImportedKeyVerificationParty =
+pub type DWalletDKGFirstParty = <AsyncProtocol as Protocol>::EncryptionOfSecretKeyShareRoundParty;
+pub(crate) type DWalletImportedKeyVerificationParty =
     <AsyncProtocol as Protocol>::TrustedDealerDKGDecentralizedParty;
 /// This struct represents the final round of the DKG protocol.
-pub(super) type DKGSecondParty = <AsyncProtocol as Protocol>::ProofVerificationRoundParty;
+pub(crate) type DWalletDKGSecondParty = <AsyncProtocol as Protocol>::ProofVerificationRoundParty;
+
+pub(crate) fn dwallet_dkg_first_public_input(
```

Comment:
It's much better to have each module responsible for its own public input and session info.